### DISCUSSION TITLE
Can't build WASM project duo to read-only asset files

### DISCUSSION THREAD

**Discussion by VRage** [unoplatform/Uno.Wasm.Bootstrap#813] [Q&A] [answered]
We are using Perforce as our revision management tool, which makes all submitted files read-only. Normally, it is not a big deal, but building a Uno WebAssembly (Wasm) project fails:

Uno.Wasm.Bootstrap.targets(207,3): error : System.UnauthorizedAccessException: Access to the path 'back.svg' is denied.

Other files are affected as well:

.Wasm/libman.json
.Wasm/wwwroot/staticwebapp.config.json
Wasm/wwwroot/web.config
/Assets/Icons/*.svg
/Assets/Icons/SharedAssets.md
It seems that during the build, the Uno WebAssembly Bootstrap copies all the files with the read-only property enabled. Subsequent builds attempt to overwrite those read-only files, leading to this error. If I manually remove the read-only property on those files in the bin/obj folder, the next build will be successful. This problem does not occur when building the Windows project.

Turning off the read-only property on every Perforce client is not an option.

**Comment by jeromelaban** [comment, marked as answer]
@VRage we made related changes after you reported this behavior, is it still happening for you?

**Reply by VRage** [reply]
@jeromelaban i've just tested  Uno.Wasm.Bootstrap 8.0.8 and it works just fine. thank you.
